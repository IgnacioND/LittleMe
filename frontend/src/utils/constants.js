export const MONEY_INCREMENT = 10;
export const STANDARD_INCREMENT = 1;
export const MONEYPLUS_INCREMENT = MONEY_INCREMENT * 2;
export const NULL_INCREMENT = 0;
