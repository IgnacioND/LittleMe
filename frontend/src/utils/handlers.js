// import { useDispatch } from 'react-redux';

export const goToTasksScreen = (navigation) => () => navigation.navigate('TasksScreen');
export const goToLoginScreen = (navigation) => () => navigation.navigate('LoginScreen');
export const goToSignupScreen = (navigation) => () => navigation.navigate('SignupScreen');
export const goToDashboardScreen = (navigation) => () => navigation.navigate('DashboardScreen');
export const goToProfileScreen = (navigation) => () => navigation.navigate('ProfileScreen');
export const goToCalendarScreen = (navigation) => () => navigation.navigate('CalendarScreen');
export const goToCharacterScreen = (navigation) => () => navigation.navigate('CharacterScreen');
export const goToCreateTaskScreen = (navigation) => () => navigation.navigate('CreateTaskScreen');
export const goToHomeScreen = (navigation) => () => navigation.navigate('HomeScreen');
export const goToScreen = (navigation) => () => navigation.navigate('Screen');
