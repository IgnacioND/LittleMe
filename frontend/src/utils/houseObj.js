const houseObj = [
  {
    name: 1,
    leftWall: 1,
    rightWall: 1,
    centerWall: 1,
    floor: 1,
    ceiling: 1,
    leftFurniture: 1,
    centerFurniture: 1,
    rightFurniture: 1,
    tidynessLevel: 1,
  },
];

module.exports = houseObj;
