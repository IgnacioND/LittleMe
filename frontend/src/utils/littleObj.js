const littleObj = {
  name: 'Pepe',
  age: 1,
  evolution: 1,
  skin: 1,
  appeareance: {
    eyes: 1,
    mouth: 1,
    hair: 1,
    extra1: 1,
    extra2: 1,
    height: 1,
    width: 1,
  },
  stats: {
    money: 3,
    hunger: 5,
    hygiene: 5,
    mind: 3,
  },
  clothes: {
    shirt: 1,
    pants: 1,
    shoes: 1,
  },
  tendencies: {
    studies: 1,
    healthiness: 1,
    leisure: 1,
  },
};
module.exports = littleObj;
