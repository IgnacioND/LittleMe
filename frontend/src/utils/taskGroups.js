const taskGroups = {
  work: ['reunion', 'meeting', 'colleage', 'worker', 'company',
    'coworker', 'co-worker', 'working', 'work', 'project'],
  studies: ['read', 'homework', 'book', 'learn', 'study', 'degree', 'master'],
  house: [
    'kitchen', 'living-room', 'room', 'bathroom',
    'buy', 'clean', 'cleaning', 'dormitory', 'hall',
    'cook', 'groceries', 'bleach', 'ironing', 'detergent',
    'laundry', 'dust', 'washing', 'floor', 'mop',
    'lawn', 'rubbish', 'tidy', 'vacuum', 'dishes', 'broom', 'chores'],
  leisure: ['friend', 'movie', 'movies', 'cinema', 'hangout',
    'hobby', 'holiday', 'relax', 'relaxation', 'rest', 'casual',
    'enjoy', 'travel', 'vacation', 'play', 'theater', 'hang out'],
  health: ['gym', 'dance', 'run', 'doctor', 'hair', 'football', 'play', 'dojo', 'fight',
    'lunch', 'medical', 'therapy', 'psycology', 'shrunk', 'eat', 'dinner', 'sleep',
    'breackfast', 'dinner', 'lunch'],

};
module.exports = taskGroups;
