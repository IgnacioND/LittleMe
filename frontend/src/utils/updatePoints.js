import {
  MONEY_INCREMENT,
  STANDARD_INCREMENT,
  MONEYPLUS_INCREMENT,
  NULL_INCREMENT,
} from './constants';

export default function updatePoints(task, completeUser) {
  const isWorkTask = task === 'work';
  const isHealthTask = task === 'health';
  const isLeisureTask = task === 'leisure';
  const isMindTask = ['study', 'house', 'leisure'].includes(task);
  const isHealthyTask = ['study', 'house', 'health'].includes(task);
  const isWorkStudyTask = ['study', 'work'].includes(task);
  const { stats, tendencies } = completeUser?.little || {};

  return {
    little: {
      stats: {
        money: stats?.money + (isWorkTask ? MONEYPLUS_INCREMENT : MONEY_INCREMENT),
        hunger: stats?.hunger + STANDARD_INCREMENT,
        hygiene: stats?.hygiene + (isHealthTask ? STANDARD_INCREMENT : NULL_INCREMENT),
        mind: stats?.mind + (isMindTask ? STANDARD_INCREMENT : NULL_INCREMENT),

      },
      tendencies: {
        studies: tendencies?.studies + (isWorkStudyTask ? STANDARD_INCREMENT : NULL_INCREMENT),
        healthiness: tendencies?.healthiness + isHealthyTask ? STANDARD_INCREMENT : NULL_INCREMENT,
        leisure: tendencies?.leisure + (isLeisureTask ? STANDARD_INCREMENT : NULL_INCREMENT),
      },
    },
  };
}
