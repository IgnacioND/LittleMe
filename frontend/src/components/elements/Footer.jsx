import React from 'react';
import PropTypes from 'prop-types';
import {
  View, Image, TouchableOpacity,
} from 'react-native';

import generalStyles from '../styles/generalStyles';
import footerStyles from '../styles/footerStyles';
import { goToCreateTaskScreen, goToTasksScreen } from '../../utils/handlers';

const iconAddTask = require('../../assets/images/iconAddTask.png');
const iconList = require('../../assets/images/iconList.png');

export default function Footer({ navigation }) {
  return (
    <View style={footerStyles.footerContainer}>
      <View style={generalStyles.iconButtonContainer}>
        <View style={generalStyles.iconButtonBack} />
        <TouchableOpacity
          style={generalStyles.iconButton}
          onPress={goToCreateTaskScreen(navigation)}
        >
          <Image source={iconAddTask} />
        </TouchableOpacity>
      </View>
      <View style={generalStyles.iconButtonContainer}>
        <View style={generalStyles.iconButtonBack} />
        <TouchableOpacity style={generalStyles.iconButton} onPress={goToTasksScreen(navigation)}>
          <Image source={iconList} />
        </TouchableOpacity>
      </View>

    </View>
  );
}

Footer.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
