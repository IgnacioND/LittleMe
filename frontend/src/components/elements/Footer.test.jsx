import React from 'react';
import { render } from '../../utils/test.utils';
import Footer from './Footer';

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Footer Component', () => {
  describe('When it is rendered', () => {
    let screen;
    beforeEach(() => {
      screen = render(<Footer navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
  describe('When Add task is clicked', () => {
    test('Should show keyboard', () => {

    });
  });
  describe('When close is clicked', () => {
    test('Should hide keyboard', () => {
    });
    test('Should not add task', () => {

    });
  });
  describe('When ok task is clicked', () => {
    describe('And there is nothing written', () => {
      test('Should not add task', () => {

      });
    });
    describe('And there is something written', () => {
      test('Should add task', () => {

      });
      test('Should hide keyboard', () => {

      });
    });
  });
});
describe('When task list is clicked', () => {
  test('Should go to tasklist', () => {

  });
});
