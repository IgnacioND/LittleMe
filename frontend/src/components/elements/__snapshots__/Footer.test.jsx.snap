// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Footer Component When it is rendered Then screen should match the snapshot 1`] = `
<View
  style={
    Object {
      "bottom": 40,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "darkgrey",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 40,
          "marginLeft": 3,
          "marginTop": 3,
          "width": 40,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFCA51",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 40,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "darkgrey",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 40,
          "marginLeft": 3,
          "marginTop": 3,
          "width": 40,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFCA51",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 40,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
</View>
`;
